datalad/distribution/get.py

```diff
@@ -243,7 +243,15 @@ def _get_flexible_source_candidates_for_submodule(ds, sm):
                         ds_repo.config[c])
                        for c in ds_repo.config.keys()
                        if c.startswith(candcfg_prefix)]:
-        url = tmpl.format(**sm_candidate_props)
+        try:
+            url = tmpl.format(**sm_candidate_props)
+        except KeyError:
+            # don't crash if the submodule lacks properties required by the
+            # template - this can happen if a pure Git repository is queried
+            # for a dataset id, for example.
+            lgr.debug('Caught a key error in the generation of a submodule '
+                      'URL using the template {}'.format(tmpl))
```

Comment:
Minor OPT: don't use format for logging messages, use % interpolation and pass arguments to debug call. See other invocations of logging for examples

Reply:
Also it is usually a good manner to log exception itself, helps to debug

Reply:
thanks for the tips!